compaction/compaction.cc

```diff
@@ -1576,7 +1623,8 @@ class resharding_compaction final : public compaction {
 future<compaction_result> compaction::run(std::unique_ptr<compaction> c) {
     return seastar::async([c = std::move(c)] () mutable {
         c->setup().get();
-        auto consumer = c->consume();
+        auto cts = seastar::make_lw_shared<compaction_manager::task::compaction_task_stats>();
```

Comment:
Does this have to be shared pointer?

I think `compaction` should create this and store it as a member. It is less noise.